### ISSUE TITLE
AccessDeniedException happen when building the image

### ISSUE THREAD

**Issue by HuangAo** [GoogleContainerTools/jib#1763]
**Description of the issue**:
when i build a springboot project on local win10 enviroment, encounter AcceessDeniedException, and i checked there's no folder named "805c6914704cef9e79074e36853298287ad1262f368d8648d52ff307dbebcf28" under target\jib-cache\layers mentioned in the ERROR log
```
[DEBUG] TIMING  Building and pushing image
[INFO]  TIMING  Retrieving registry credentials for registry-1.docker.io
[INFO]  TIMING  Pulling base image manifest
[INFO]  TIMING  Building application layers
[INFO]  TIMED   Building application layers : 1.0 ms
[INFO]  TIMING  Building classes layer
[INFO]  TIMING  Building dependencies layer
[INFO]  TIMING  Building extra files layer
[INFO]  TIMING  Building snapshot dependencies layer
[INFO]  TIMING  Building resources layer
[INFO]  TIMING  Setting up to push layers
[INFO]  TIMED   Setting up to push layers : 2.0 ms
[INFO]  TIMED   Building snapshot dependencies layer : 647.0 ms
[INFO]  TIMED   Building extra files layer : 656.0 ms
[INFO]  Building resources layer built sha256:24785422b8265b79eb17fce1274dda5622fd8f97e5baf233aef635367a122547
[INFO]  TIMED   Building resources layer : 1744.0 ms
[INFO]  Building classes layer built sha256:2c4829a880a4958ac7272e9101e967e764c7fd76aaefb9cae2ce4436481f94f4
[INFO]  TIMED   Building classes layer : 1853.0 ms
[INFO]  Cannot verify server at https://127.0.0.1:5000/v2/openjdk11/manifests/alpine-jre. Attempting again with no TLS verification.
[INFO]  Failed to connect to https://127.0.0.1:5000/v2/openjdk11/manifests/alpine-jre over HTTPS. Attempting again with HTTP: http://127.0.0.1:5000/v2/openjdk11/manifests/alpine-jre
[INFO]  Cannot verify server at https://127.0.0.1:5000/v2/openjdk11/blobs/sha256:37888a81f3758f5726758ffa035b649656c56899effdca7fe941209041620b41. Attempting again with no TLS verification.
[INFO]  Failed to connect to https://127.0.0.1:5000/v2/openjdk11/blobs/sha256:37888a81f3758f5726758ffa035b649656c56899effdca7fe941209041620b41 over HTTPS. Attempting again with HTTP: http://127.0.0.1:5000/v2/openjdk11/blobs/sha256:37888a81f3758f5726758ffa035b649656c56899effdca7fe941209041620b41
[INFO]  TIMED   Pulling base image manifest : 2386.0 ms
[INFO]  TIMING  Setting up base image caching
[INFO]  TIMED   Setting up base image caching : 1.0 ms
[INFO]  TIMING  Setting up to push layers
[INFO]  TIMING  Pulling base image layer sha256:e7c96db7181be991f19a9fb6975cdbbd73c65f4a2681348e63a141a2192a5f10
[INFO]  TIMED   Setting up to push layers : 0.0 ms
[INFO]  TIMING  Pulling base image layer sha256:8532d8865aed201cb687af15ec1c8e8b163a32a03ca3e91f02fd5e57139ad975
[INFO]  TIMING  Pulling base image layer sha256:8f768d60fb269e863f8ccef1cddc78d1e1b9371a26b5a3a0c090b02c63f965e9
[INFO]  TIMED   Pulling base image layer sha256:8532d8865aed201cb687af15ec1c8e8b163a32a03ca3e91f02fd5e57139ad975 : 2.0 ms
[INFO]  TIMED   Pulling base image layer sha256:e7c96db7181be991f19a9fb6975cdbbd73c65f4a2681348e63a141a2192a5f10 : 2.0 ms
[INFO]  TIMED   Pulling base image layer sha256:8f768d60fb269e863f8ccef1cddc78d1e1b9371a26b5a3a0c090b02c63f965e9 : 2.0 ms
[INFO]  Using credentials from Docker config for registry-1.docker.io
[INFO]  TIMED   Retrieving registry credentials for registry-1.docker.io : 2487.0 ms
[INFO]  TIMING  Authenticating with push to registry-1.docker.io
[INFO]  Failed to connect to https://registry-1.docker.io/v2/ over HTTPS. Attempting again with HTTP: http://registry-1.docker.io/v2/
[INFO]  TIMED   Building dependencies layer : 12114.0 ms
[INFO]  TIMED   Building and pushing image : 12201.0 ms
[INFO] Executing tasks:
[INFO] [============                  ] 40.0% complete
[INFO] > setting up to push layers
[INFO] > setting up to push layers
[INFO] > authenticating push to registry-1.docker.io
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 42.457 s
[INFO] Finished at: 2019-06-06T10:15:16+08:00
[INFO] Final Memory: 65M/515M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.2.0:build (default-cli) on project noapi: C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache\tmp\448194133844108686 -> C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache\layers\805c6914704cef9e79074e36853298287ad1262f368d8648d52ff307dbebcf28 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.2.0:build (default-cli) on project noapi: C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache\tmp\448194133844108686 -> C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache\layers\805c6914704cef9e79074e36853298287ad1262f368d8648d52ff307dbebcf28
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache\tmp\448194133844108686 -> C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache\layers\805c6914704cef9e79074e36853298287ad1262f368d8648d52ff307dbebcf28
        at com.google.cloud.tools.jib.maven.BuildImageMojo.execute(BuildImageMojo.java:168)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 20 more
Caused by: java.nio.file.AccessDeniedException: C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache\tmp\448194133844108686 -> C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache\layers\805c6914704cef9e79074e36853298287ad1262f368d8648d52ff307dbebcf28
        at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:83)
        at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
        at sun.nio.fs.WindowsFileCopy.move(WindowsFileCopy.java:387)
        at sun.nio.fs.WindowsFileSystemProvider.move(WindowsFileSystemProvider.java:287)
        at java.nio.file.Files.move(Files.java:1395)
        at com.google.cloud.tools.jib.cache.CacheStorageWriter.moveIfDoesNotExist(CacheStorageWriter.java:85)
        at com.google.cloud.tools.jib.cache.CacheStorageWriter.writeUncompressed(CacheStorageWriter.java:225)
        at com.google.cloud.tools.jib.cache.Cache.writeUncompressedLayer(Cache.java:115)
        at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call(BuildAndCacheApplicationLayerStep.java:138)
        at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call(BuildAndCacheApplicationLayerStep.java:40)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Steps to reproduce**:
run "mvn package -Pdev package jib:dockerBuild"

**Environment**: 
win10, DockerDesktop, jib-maven-plugin 1.2.0
**`jib-maven-plugin` Configuration**: 
```xml
<plugin>
                   <groupId>com.google.cloud.tools</groupId>
                   <artifactId>jib-maven-plugin</artifactId>
                   <version>1.2.0</version>
                   <configuration>
                       <from>
                           <image>127.0.0.1:5000/openjdk11:alpine-jre</image>
                       </from>
                       <to>
                           <image>myappliction:latest</image>
                       </to>
                       <container>
                           <entrypoint>
                               <shell>sh</shell>
                               <option>-c</option>
                               <arg>chmod +x /entrypoint.sh &amp;&amp; sync &amp;&amp; /entrypoint.sh</arg>
                           </entrypoint>
                           <ports>
                               <port>8081</port>
                               <port>5701/udp</port>
                           </ports>
                           <environment>
                               <SPRING_OUTPUT_ANSI_ENABLED>ALWAYS</SPRING_OUTPUT_ANSI_ENABLED>
                               <JHIPSTER_SLEEP>0</JHIPSTER_SLEEP>
                           </environment>
                           <useCurrentTimestamp>true</useCurrentTimestamp>
                       </container>
                       <allowInsecureRegistries>true</allowInsecureRegistries>
                   </configuration>
               </plugin>
```

**Comment by chanseokoh** [comment]
Hi @HuangAo,

Is this error always reproducible? We occasionally get these exception reports only on Windows, and often this is due to anti-virus software briefly grabbing a hold of the directory for examination.

https://github.com/GoogleContainerTools/jib/issues/1438
https://github.com/GoogleContainerTools/jib/issues/1635#issuecomment-484199610

Try disabling any kind of process or anti-virus software.

Unrelated,
```
<arg>chmod +x /entrypoint.sh
```
it's possible to set file permissions through `<extraDirectories><permissions>`. See [an example](https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#adding-arbitrary-files-to-the-image).

**Comment by briandealwis** [comment]
@HuangAo is there any kind of sharing or replication happening with this file path?  I'm not a Windows user, but I remember there is some way to share files and directories as part of a roaming profile?  Our is this path being shared with a Docker container or a VM?  
```
C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache
```

**Comment by HuangAo** [comment]
> @HuangAo is there any kind of sharing or replication happening with this file path? I'm not a Windows user, but I remember there is some way to share files and directories as part of a roaming profile? Or is this path being shared with a Docker container or a VM?
> 
> ```
> C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache
> ```

Hi @briandealwis ,
there's no sharing or replication with this file path, every time after i run the build command, there will be 4 layers build successfully under C:\Users\huangao\Desktop\Docker\test\Noapi\target\jib-cache\layers, the error happened in building the layer ''805c6914704cef9e79074e36853298287ad1262f368d8648d52ff307dbebcf28'' everytime

**Comment by chanseokoh** [comment]
No clue really. Sounds like this happens all the time. Is it the same with `mvn clean ...` to start fresh?

Another thing we can try is to change the application cache directory to a completely different place and see if there is any difference using `-Djib.applicationCache=...` (https://github.com/GoogleContainerTools/jib/issues/1635#issuecomment-484199610), maybe like `C:\Usres\huangao\jib-cache` or even a different disk/filesystem? Lastly, although I don't think this matters, also try combining that with `-DjibSerialize=true` (note no period between `jib` and `Serialize`).

**Comment by briandealwis** [comment]
Hi @HuangAo. We recently introduced a change with #1887 that we think will address the issue you reported.  Is it possible for you to build Jib from HEAD and give it a try?  You should be able to install it to your local repository with something like the following:
```
$ cd jib-maven-plugin
$ mvn install
...
$ cd /your/other/project
$ mvn -Djib.version=1.4.1-SNAPSHOT package
```

**Comment by chanseokoh** [comment]
@HuangAo we just released 1.5.0 that has the fix @briandealwis was talking about. (It will retry file I/O a few more times in the case of a failure.) Can you give it a try?

**Comment by chanseokoh** [comment]
Closing this due to inactivity. Hopefully this is sidestepped with the retry feature.
